mono/utils/mono-merp.c

```diff
@@ -471,6 +488,15 @@ mono_merp_invoke (const intptr_t crashed_pid, const char *signal, const char *no
 	return success;
 }
 
+void
+mono_merp_add_annotation (const char *key, const char *value)
+{
+	MonoMerpAnnotationEntry *entry = g_new0 (MonoMerpAnnotationEntry, 1);
+	entry->key = g_strdup (key);
+	entry->value = g_strdup (value);
+	config.annotations = g_list_append (config.annotations, entry);
```

Comment:
How many annotations are you expecting?  This is an O(n) insertion.  Could use a `GQueue` to get constant-time appends.

Reply:
Good catch. I think I'll just prepend to an slist.

Reply:
A queue to me implies that we care about the FIFO behavior. This is going to be immutable after it's installed.